modules/apps/27-interchain-accounts/types/account.go

```diff
@@ -11,71 +10,52 @@ import (
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 	yaml "gopkg.in/yaml.v2"
+)
 
-	connectiontypes "github.com/cosmos/ibc-go/v2/modules/core/03-connection/types"
+var (
+	_ authtypes.GenesisAccount = (*InterchainAccount)(nil)
+	_ InterchainAccountI       = (*InterchainAccount)(nil)
 )
 
-// GenerateAddress returns an sdk.AccAddress derived using the provided module account address and port identifier.
-// The sdk.AccAddress returned is a sub-address of the module account, using the controller chain's port identifier as the derivation key
-func GenerateAddress(moduleAccAddr sdk.AccAddress, portID string) sdk.AccAddress {
-	return sdk.AccAddress(sdkaddress.Derive(moduleAccAddr, []byte(portID)))
+// InterchainAccountI wraps the authtypes.AccountI interface
+type InterchainAccountI interface {
+	authtypes.AccountI
 }
 
-// GeneratePortID generates the portID for a specific owner
-// on the controller chain in the format:
-//
-// 'ics-27-<connectionSequence>-<counterpartyConnectionSequence>-<owner-address>'
-// https://github.com/seantking/ibc/tree/sean/ics-27-updates/spec/app/ics-027-interchain-accounts#registering--controlling-flows
-// TODO: update link to spec
-func GeneratePortID(owner, connectionID, counterpartyConnectionID string) (string, error) {
```

Comment:
- Moved `GeneratePortID` to port.go
- Moved interface/struct definitions to top of file
- Added/updated godocs